include/range/v3/utility/array.hpp

```diff
@@ -0,0 +1,304 @@
+            const value_type* data() const noexcept
+            {
+                return elems_;
+            }
+        };
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator==(const array<T, N>& x, const array<T, N>& y)
+        {
+            return ranges::equal(x.elems_, x.elems_ + N, y.elems_);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator!=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(x == y);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator<(const array<T, N>& x, const array<T, N>& y)
+        {
+            return ranges::lexicographical_compare(x.elems_, x.elems_ + N, y.elems_, y.elems_ + N);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator>(const array<T, N>& x, const array<T, N>& y)
+        {
+            return y < x;
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator<=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(y < x);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        bool
+        operator>=(const array<T, N>& x, const array<T, N>& y)
+        {
+            return !(x < y);
+        }
+
+        template <class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        auto swap(const array<T, N>& x, const array<T, N>& y)
+                noexcept(is_nothrow_swappable<T, T>::value)
+                -> typename std::enable_if<is_swappable<T, T>::value, void>::type
+        {
+            x.swap(y);
+        }
+
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        T& get(array<T, N>& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (ranges::array)");
+            return a.elems_[I];
+        }
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        const T& get(const array<T, N>& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (const ranges::array)");
+            return a.elems_[I];
+        }
+
+        template <size_t I, class T, size_t N>
+        RANGES_CXX14_CONSTEXPR
+        T&& get(array<T, N>&& a) noexcept
+        {
+            static_assert(I < N, "Index out of bounds in ranges::get<> (ranges::array &&)");
+            return std::move(a.elems_[I]);
+        }
+
+    }  // namespace v3
+}  // namespace ranges
+
+namespace std
+{
+
+template <class T, size_t N>
+class tuple_size<ranges::array<T, N>>
+    : public integral_constant<size_t, N> {};
+
+template <size_t I, class T, size_t N>
+class tuple_element<I, ranges::array<T, N> >
+{
+ public:
+    using type = T;
+};
+
+}  // namespace std
+
+#endif // RANGES_V3_UTILITY_ARRAY_HPP
```

Comment:
I would suggest a minimal constexpr-friendly `array` container in the test directory. Containers don't belong in range-v3 proper.